### PR TITLE
changefeedccl: add tests for changefeeds on enum types

### LINKED ISSUE #49981: sql: investigate interaction between user defined types and changefeeds
At the very least we need some tests here to ensure that everything works as expected.

### PR DESCRIPTION
**Pull request by rohany**
Work for #49981.

This PR adds a basic test for a changefeed on a table with an enum type.

Release note: None

### CODE REVIEW

**Review by ajwerner — APPROVED** [review]
This :lgtm: but I think there's more testing to be done here. 

There's going to be more work that we're going to need to do here when the leasing comes in. Today the changefeed keeps track of the set of descriptors that it is watching and ensure that when it sees a row that the lease manager has the most up-to-date version of the descriptor for that row. It does that by polling the descriptor table and waiting for that polling. When the polling picks up a new version it enforces that the lease manager pick up that newest version. I think ultimately we're going to want `AcquireFreshestFromStore` to potentially recurse into calling the same for each of referenced types. See https://github.com/cockroachdb/cockroach/blob/master/pkg/ccl/changefeedccl/schemafeed/schema_feed.go#L417-L426

In short, it will be bad if there's an update to the set of values for an enum and we don't see that reflected in the hydrated, leased table descriptor. 

I think what I'd love to see is an extension to the nemesis to use enum types and add new values and use those new values. I'd expect that if you just did that at some point we'd hit a case where we find a row with an enum value we don't know how to map into its string. Will we be able to hit that today? I don't remember whether we bump the versions of tables which reference a type.

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50097#-:-M9Z4vCJ25KjnibiwlSV:bapp9yj)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @pbardea)

**Comment by rohany** [comment]
I see, thanks for the insight! I'll change this to keep the issue open, and note to update the nemesis test once we can change change and lease types.

bors r=ajwerner

**Comment by rohany** [comment]
Writing this down so I remember:

Changefeeds get table descriptors to decode keys with from the `rowFetcherCache`, which uses the lease manager to get tables. The test happened to be passing due to some sharing of type metadata which happens only because we are temporarily hydrating type metadata in the planner on immutable table descriptors. When the lease manager needs to actually go to disk, it would fail. This is because the lease manager doesn't do type hydration yet, and eventually will need to. 

I'm going to put this PR aside for now until we have type leasing.
